### PR TITLE
Add Pot monad + react utilities.

### PR DESCRIPTION
**Pull request by rpiaggio**

### CODE REVIEW

File: shared/src/main/scala/crystal/data/implicits/package.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package crystal.data
+
+import cats.kernel.Eq
+import cats.implicits._
+
+package object implicits {
+  implicit val eqThrowable: Eq[Throwable] = Eq.by(_.getClass.getName)
```
Comment (cquiroz) [inline shared/src/main/scala/crystal/data/implicits/package.scala:7]: This is suspicious. Have you seen?
https://github.com/typelevel/cats-effect/issues/271

File: shared/src/main/scala/crystal/data/Pot.scala [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            }
+          case Ready(ax)       =>
+            y match {
+              case Ready(ay) => ax === ay
+              case _         => false
+            }
+        }
+    }
+
+  implicit object PotMonad extends Monad[Pot] {
```
Comment (cquiroz) [inline shared/src/main/scala/crystal/data/Pot.scala:70]: Nice, Do you think we can also implement for example `Traverse` or `Foldable`?

File: shared/src/main/scala/crystal/data/implicits/package.scala (hunk 1 above)
Comment (rpiaggio) [reply shared/src/main/scala/crystal/data/implicits/package.scala:7]: I wasn't very sure about this and hadn't seen that discussion. I'll make all `Throwable`s `Eq`ual then.

BTW, we also have code where `Reusability[Throwable]` is needed somewhere else, but I think that's a different case: 

Let's say we have a component that renders a `Pot` which is in an `Error(throwable1)` state. The action is retried and the `Pot` quickly goes through a `Pending` that doesn't make it to the rendering component, and ends up in a new `Error(throwable2)` state. Do you think it's worth it to implement a generic `Reusable[Throwable]` that compares probably the class name and message? Or should we leave it open for the client to specify?

File: shared/src/main/scala/crystal/data/Pot.scala (hunk 2 above)
Comment (rpiaggio) [reply shared/src/main/scala/crystal/data/Pot.scala:70]: I'll look into it.

File: shared/src/main/scala/crystal/data/implicits/package.scala (hunk 1 above)
Comment (cquiroz) [reply shared/src/main/scala/crystal/data/implicits/package.scala:7]: Yeah. I'd leave it open. You could add instances but not make them implicit and let the client reuse them if they want

**Pull request merged**